react/react1/week3/my-todo-list/src/components/TodoList.js

```diff
@@ -0,0 +1,68 @@
+import { useState, useEffect } from 'react';
+import Button from './Button';
+import TodoRow from './TodoRow';
+import AddTodo from './AddTodo';
+import Calendar from './Calendar';
+import BlackBorder from './BorderComponent';
+
+const TodoList = () => {
+  const [date, setDate] = useState('');
+  const [newTodo, setNewTodo] = useState('');
+  const [todosList, setTodosList] = useState([]);
+
+  const API_URL =
+    'https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw';
+
+  const FetchApi = () => {
+    fetch(API_URL)
+      .then((response) => response.json())
+      .then((result) => setTodosList(result));
+  };
+
+  useEffect(() => {
+    FetchApi();
+  }, []);
+
+  let uniqId = Math.random().toString(16).slice(2);
+
+  const newTodoItem = {
+    id: uniqId,
+    deadline: date,
+    description: newTodo,
+  };
+
+  function addTodo() {
+    setTodosList([...todosList, newTodoItem]);
+  }
+
+  function deleteTodoList(id) {
+    const updetedList = todosList.filter((el) => el.id !== id);
+
+    setTodosList(updetedList);
+  }
+
+  return (
+    <>
+      <AddTodo newTodo={newTodo} setNewTodo={setNewTodo} />
+      <Calendar date={date} setDate={setDate} />
+      <Button onClick={addTodo} bigSize text='Add todo' />
+      <div>
+        {todosList.map((todo) => (
+          <BlackBorder>
+            <TodoRow
+              todo={todo}
+              key={todo.id}
+              todosList={todosList}
```

Comment:
The ToDoRow is actually unneccesary you have allready dividid your application into many small components. So what and what you end up doing here is having two centralised components. ToDoList and ToDoRow. So instead of passing all these props down to the TodoRow component i would advise you to consider a way to construct this in a way where you dont pass 10 different props to a new component. This makes the code harder to debug and less maintainable, so try looking into a concept called component composition.